File: Claude-T2/apps/api-java/src/main/java/com/lifeline/modules/role/service/RoleService.java
package com.lifeline.modules.role.service;

import com.lifeline.common.exception.BusinessException;
import com.lifeline.modules.role.dto.CreateRoleDto;
import com.lifeline.modules.role.dto.UpdateRoleDto;
import com.lifeline.modules.role.entity.Role;
import com.lifeline.modules.role.repository.RoleRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Service
@RequiredArgsConstructor
public class RoleService {

    private final RoleRepository roleRepository;

    public List<Role> findAll() {
        return roleRepository.findAll(org.springframework.data.domain.Sort.by(
                org.springframework.data.domain.Sort.Direction.ASC, "createdAt"));
    }

    public Role findOne(String id) {
        return roleRepository.findById(id)
                .orElseThrow(() -> BusinessException.notFound("Role with ID \"" + id + "\" not found"));
    }

    public Role findByCode(String code) {
        return roleRepository.findByCode(code).orElse(null);
    }

    @Transactional
    public Role create(CreateRoleDto dto) {
        Role role = new Role();
        role.setName(dto.getName());
        role.setCode(dto.getCode());
        role.setDescription(dto.getDescription());
        role.setPermissions(dto.getPermissions());
        return roleRepository.save(role);
    }

    @Transactional
    public Role update(String id, UpdateRoleDto dto) {
        Role role = findOne(id);
        if (dto.getName() != null) role.setName(dto.getName());
        if (dto.getDescription() != null) role.setDescription(dto.getDescription());
        if (dto.getPermissions() != null) role.setPermissions(dto.getPermissions());
        return roleRepository.save(role);
    }

    @Transactional
    public void remove(String id) {
        Role role = findOne(id);
        roleRepository.delete(role);
    }

    @Transactional
    public void seedDefaultRoles() {
        seedIfNotExists("管理员", "admin", "系统管理员，拥有所有权限", new String[]{"*"});
        seedIfNotExists("运维员", "operator", "运维人员，可操作设备和处理预警",
                new String[]{"device:read", "device:write", "alert:read", "alert:write", "telemetry:read"});
        seedIfNotExists("观察员", "observer", "观察员，只读权限",
                new String[]{"device:read", "alert:read", "telemetry:read"});
    }

    private void seedIfNotExists(String name, String code, String desc, String[] perms) {
        if (roleRepository.findByCode(code).isEmpty()) {
            Role role = new Role();
            role.setName(name);
            role.setCode(code);
            role.setDescription(desc);
            role.setPermissions(perms);
            roleRepository.save(role);
        }
    }
}

File: Claude-T2/apps/api/src/modules/role/role.service.ts
import { Injectable, NotFoundException } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { Role, RoleCode } from './role.entity';

@Injectable()
export class RoleService {
  constructor(
    @InjectRepository(Role)
    private readonly roleRepository: Repository<Role>,
  ) {}

  async findAll(): Promise<Role[]> {
    return this.roleRepository.find({
      order: { createdAt: 'ASC' },
    });
  }

  async findOne(id: string): Promise<Role> {
    const role = await this.roleRepository.findOne({ where: { id } });
    if (!role) {
      throw new NotFoundException(`Role with ID "${id}" not found`);
    }
    return role;
  }

  async findByCode(code: RoleCode): Promise<Role | null> {
    return this.roleRepository.findOne({ where: { code } });
  }

  async create(roleData: Partial<Role>): Promise<Role> {
    const role = this.roleRepository.create(roleData);
    return this.roleRepository.save(role);
  }

  async update(id: string, roleData: Partial<Role>): Promise<Role> {
    const role = await this.findOne(id);
    Object.assign(role, roleData);
    return this.roleRepository.save(role);
  }

  async remove(id: string): Promise<void> {
    const role = await this.findOne(id);
    await this.roleRepository.remove(role);
  }

  async seedDefaultRoles(): Promise<void> {
    const defaultRoles = [
      {
        name: '管理员',
        code: RoleCode.ADMIN,
        description: '系统管理员，拥有所有权限',
        permissions: ['*'],
      },
      {
        name: '运维员',
        code: RoleCode.OPERATOR,
        description: '运维人员，可操作设备和处理预警',
        permissions: ['device:read', 'device:write', 'alert:read', 'alert:write', 'telemetry:read'],
      },
      {
        name: '观察员',
        code: RoleCode.OBSERVER,
        description: '观察员，只读权限',
        permissions: ['device:read', 'alert:read', 'telemetry:read'],
      },
    ];

    for (const roleData of defaultRoles) {
      const existing = await this.findByCode(roleData.code);
      if (!existing) {
        await this.create(roleData);
      }
    }
  }
}
